extensions/amp-byside-placeholder/0.1/amp-byside-placeholder.js

```diff
@@ -0,0 +1,348 @@
+        (this.element.getAttribute('data-webcare-id') ||
+        this.element.getAttribute('webcare-id')),
+        'The data-webcare-id attribute is required for <' + TAG_ + '> %s',
+        this.element);
+
+    this.label_ = user().assert(
+        (this.element.getAttribute('data-label') ||
+        this.element.getAttribute('label')),
+        'The data-label attribute is required for <' + TAG_ + '> %s',
+        this.element);
+
+    this.channel_ = (this.element.getAttribute('data-channel') ||
+        this.element.getAttribute('channel') || '');
+    this.lang_ = (this.element.getAttribute('data-lang') ||
+            this.element.getAttribute('lang') || DEFAULT_LANG_);
+    this.fid_ = (this.element.getAttribute('data-fid') ||
+            this.element.getAttribute('fid') || '');
+
+    this.origin_ = this.generateOrigin_();
+    this.baseUrl_ = this.origin_ + '/BWA' + this.webcareId_ + '/amp/';
+
+    if (this.isResizable_) {
+      this.element.setAttribute('scrolling', 'no');
+    }
+
+    if (!this.element.hasAttribute('frameborder')) {
+      this.element.setAttribute('frameborder', '0');
+    }
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    const placeholder = this.win.document.createElement('div');
+    placeholder.setAttribute('placeholder', '');
+    const image = this.win.document.createElement('amp-img');
+    image.setAttribute('noprerender', '');
+    image.setAttribute('layout', 'fill');
+    image.setAttribute('referrerpolicy', 'origin');
+
+    this.propagateAttributes(['alt'], image);
+
+    // Use custom placeholder or use default transparent placeholder image
+    const placeholderImg = this.element.getAttribute('placeholder') ||
+      this.baseUrl_ + 'images/placeholder.png';
+    image.setAttribute('src', placeholderImg);
+
+    placeholder.appendChild(image);
+    return placeholder;
+  }
+
+  /** @override */
+  layoutCallback() {
+    const iframe = this.element.ownerDocument.createElement('iframe');
+    this.iframe_ = iframe;
+
+    const width = this.element.getAttribute('width') || DEFAULT_IFRAME_WIDTH_;
+    const height = this.element.getAttribute('height') ||
+        DEFAULT_IFRAME_HEIGHT_;
+
+    iframe.name = 'amp_byside_placeholder_iframe' + iframeCount_++;
+    iframe.setAttribute('frameborder', '0');
+    iframe.setAttribute('allowtransparency', 'true');
+    iframe.setAttribute('title', this.element.getAttribute('alt') || '');
+
+    if (this.element.hasAttribute('width')) {
+      iframe.setAttribute('width', width);
+    }
+    if (this.element.hasAttribute('height')) {
+      iframe.setAttribute('height', height);
+    }
+
+    setStyles(iframe, {
+      'opacity': 0,
+    });
+
+    const self = this;
+
+    return this.generateSrcUrl_().then(src => {
+      this.iframeSrc_ = src;
+      iframe.src = this.iframeSrc_;
+
+      listenFor(iframe, 'embed-size', data => {
+        this.updateSize_(data['height'], data['width']);
+      });
+
+      this.applyFillContent(iframe);
```

Comment:
move this to before the call to `generateSrcUrl_` so it is closed to other iframe config settings